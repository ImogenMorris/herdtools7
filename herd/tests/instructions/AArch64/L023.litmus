AArch64 L023
Stable=X9
(* Inter thread function calls *)
{
    0:X8=P1:Ldec;
}
  P0          | P1           ;
 MOV W9,#4    | ADR X1,Linc  ;
 BLR X8       | MOV W9,#4    ;
 BLR X8       | BLR X1       ;
 MOV W0,W9    | BLR X1       ;
 B Lzero      | MOV W0,W9    ;
Linc:         | B Lone       ;
 ADD W9,W9,#1 |Ldec:         ;
 RET          | SUB W9,W9,#1 ;
Lzero:        | RET          ;
              |Lone:         ;
forall 0:X0=2 /\ 1:X0=6
