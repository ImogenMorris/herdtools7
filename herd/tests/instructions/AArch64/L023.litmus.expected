Test L023 Required
States 1
0:X0=2; 1:X0=6;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=2 /\ 1:X0=6)
Observation L023 Always 1 0
Hash=14f65c1615cf4aacb9a537cc98e98f15

